machine/v1/types_nutanixprovider.go

```diff
@@ -154,6 +165,139 @@ type NutanixResourceIdentifier struct {
+// +kubebuilder:validation:Enum=On;Off
+type NutanixDiskFlashMode string
+
+const (
+	// NutanixDiskFlashModeOn represents the disk flash mode is On.
+	NutanixDiskFlashModeOn NutanixDiskFlashMode = "On"
+
+	// NutanixDiskFlashModeOff represents the disk flash mode is Off.
+	NutanixDiskFlashModeOff NutanixDiskFlashMode = "Off"
+)
+
+// NutanixVMStorageConfig specifies the storage configuration parameters for VM disks.
+type NutanixVMStorageConfig struct {
+	// flashMode specifies whether to pin the VM disk to the flash tier.
+	// +kubebuilder:default=Off
+	FlashMode NutanixDiskFlashMode `json:"flashMode"`
+
+	// storageContainer refers to the storage_container used by the VM disk.
+	// Currently the storageContainer configuration is supported only by using its UUID.
+	// In the future, we may support to identify storageContainer by its name.
+	// +optional
+	StorageContainer *NutanixResourceIdentifier `json:"storageContainer"`
+}
+
+// NutanixDiskDeviceType is the VM disk device type.
+// +kubebuilder:validation:Enum=Disk;CDRom
+type NutanixDiskDeviceType string
+
+const (
+	// NutanixDiskDeviceTypeDisk specifies the VM disk device type is DISK.
+	NutanixDiskDeviceTypeDisk NutanixDiskDeviceType = "Disk"
+
+	// NutanixDiskDeviceTypeCDROM specifies the VM disk device type is CDROM.
+	NutanixDiskDeviceTypeCDROM NutanixDiskDeviceType = "CDRom"
+)
+
+// NutanixDiskAdapterType is an enumeration of different disk device adapter types.
+// +kubebuilder:validation:Enum:=SCSI;IDE;PCI;SATA;SPAPR
+type NutanixDiskAdapterType string
+
+const (
+	// NutanixDiskAdapterTypeSCSI represents the disk adapter type "SCSI".
+	NutanixDiskAdapterTypeSCSI NutanixDiskAdapterType = "SCSI"
+
+	// NutanixDiskAdapterTypeIDE represents the disk adapter type "IDE".
+	NutanixDiskAdapterTypeIDE NutanixDiskAdapterType = "IDE"
+
+	// NutanixDiskAdapterTypePCI represents the disk adapter type "PCI".
+	NutanixDiskAdapterTypePCI NutanixDiskAdapterType = "PCI"
+
+	// NutanixDiskAdapterTypeSATA represents the disk adapter type "SATA".
+	NutanixDiskAdapterTypeSATA NutanixDiskAdapterType = "SATA"
+
+	// NutanixDiskAdapterTypeSPAPR represents the disk adapter type "SPAPR".
+	NutanixDiskAdapterTypeSPAPR NutanixDiskAdapterType = "SPAPR"
+)
+
+// NutanixVMDiskDeviceProperties specifies the disk device type and disk address.
+type NutanixVMDiskDeviceProperties struct {
+	// deviceType is a disk type. The default is DISK.
+	// +kubebuilder:default=Disk
+	// +kubebuilder:validation:Required
+	DeviceType NutanixDiskDeviceType `json:"deviceType"`
+
+	// adapterType is the adapter type of the disk address.
+	// If the deviceType is DISK, the adapter type can be SCSI, IDE, PCI, SATA or SPAPR.
+	// If the deviceType is CDROM, the adapter type can be IDE or SATA.
+	// +kubebuilder:default=SCSI
+	// +kubebuilder:validation:Required
+	AdapterType NutanixDiskAdapterType `json:"adapterType,omitempty"`
+
+	// deviceIndex is the index of the disk address.
```

Comment:
Explain that this must be a positive, and, is there a maximum?
```suggestion
	// deviceIndex is the index of the disk address.
	// This must be a non-negative integer value between 0 and xxx inclusive.
```

Reply:
Changed and added the godoc explanation.